core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -229,6 +229,16 @@ class SocketServer(val config: KafkaConfig, val metrics: Metrics, val time: Time
     }
   }
 
+  def updateMaxConnectionsPerIp(maxConnectionsPerIp: Int): Unit = synchronized {
```

Comment:
Should we update the fields in `SocketServer` as well?

Also, can you explain the use of this synchronization? We are protecting the updating of the max connection fields, but do we similarly protect access to these fields?

Reply:
Removed the fields from SocketServer. We can directly use config variables in startup method.

you are correct, we don't need synchronization.  Updated the PR. Adding synchronization may create unnecessary performance overhead during connection creation.